metaflow/plugins/events_decorator.py

```diff
@@ -478,7 +481,7 @@ def flow_init(
                                 )
                         if "project_branch" in flow:
                             if is_stringish(flow["project_branch"]):
-                                result["branch"] = flow["project_branch"]
+                                result["project_branch"] = flow["project_branch"]
```

Comment:
this change seems to actually fix a bug with the argo events implementation that has gone unnoticed. Currently deploying a flow with
```python
@trigger_on_finish(flow={"name": "ProjectEventsTestFlow", "project": "sa_test_project", "project_branch": "user.saikonen"})
```
does not apply a filter on the project_branch though it should.

with the introduced changes this is correctly being applied. The relevant code is in https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/argo/argo_workflows.py#L641-L646
cc @savingoyal

Reply:
I'm not sure if this is a regression or not. have triggers filtered by project&branch been working in the past with argo?